lib/model-builder.js

```diff
@@ -332,6 +339,38 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
     dataSource.attach(this);
   };
 
+  // initialize getMergePolicy if not done already via model heritance
+  ModelClass.getMergePolicy = ModelClass.getMergePolicy || function() {
+    const mergePolicy = {
+      description: {replace: true}, // string or array
+      options: {patch: true}, // object
+      properties: {patch: true}, // object
+      hidden: {replace: false}, // array
+      protected: {replace: false}, // array
+      indexes: {patch: true}, // object
+      methods: {patch: true}, // object
+      mixins: {patch: true}, // object
+      // validations: {patch: true}, // object // NOTE: not implemented
+      relations: {patch: true},
+      scope: {replace: true}, // object
+      scopes: {patch: true}, // object
+      acls: {rank: true}, // array
+      // this setting controls which child model property's value allows deleting
+      // a base model's property
+      __delete: null,
+      // this setting controls the default merge behaviour for settings not defined
+      // in the mergePolicy specification
+      __default: {replace: true},
+    };
+
+    // make sure we take settings of current model with `this`
+    // model might have no settings in case it is an instance of builtin ModelBaseClass
+    return (this.settings || {}).mergePolicy || mergePolicy;
+  };
+
+  // ModelClass.base might not have getMergePolicy() if it is an instance of builtin ModelBaseClass
+  ModelClass.base.getMergePolicy = ModelClass.base.getMergePolicy || ModelClass.getMergePolicy;
```

Comment:
ModelClass.base.getMergePolicy is assigned only if not set already. ModelClass.getMergePolicy is assigned by .extend().
 
Assigning on ModelClass.base instead of ModelClass allows to have getMergePolicy also on ModelBaseClass (model.js) when defining a class with this base class (which is default and doesn't not call .extend() internally). 
And thus extending .getMergePolicy() is possible on top of base.getMergePolicy() (see the related test).